nni/retiarii/execution/logical_optimizer/logical_plan.py

```diff
@@ -181,25 +181,25 @@ def assemble(self, multi_model_placement: Dict[Model, PhysicalDevice]) \
                 new_node, placement = node.assemble(multi_model_placement)
                 if isinstance(new_node.operation, _IOPseudoOperation):
                     model_id = new_node.graph.model.model_id
-                    if model_id not in training_config_slot:
-                        phy_model.training_config.kwargs['model_kwargs'].append(new_node.graph.model.training_config.kwargs.copy())
-                        training_config_slot[model_id] = len(phy_model.training_config.kwargs['model_kwargs']) - 1
-                        slot = training_config_slot[model_id]
-                        phy_model.training_config.kwargs['model_kwargs'][slot]['model_id'] = model_id
-                        phy_model.training_config.kwargs['model_kwargs'][slot]['use_input'] = False
-                        phy_model.training_config.kwargs['model_kwargs'][slot]['use_output'] = False
+                    if model_id not in evaluator_slot:
+                        phy_model.evaluator.kwargs['model_kwargs'].append(new_node.graph.model.evaluator.kwargs.copy())
+                        evaluator_slot[model_id] = len(phy_model.evaluator.kwargs['model_kwargs']) - 1
+                        slot = evaluator_slot[model_id]
+                        phy_model.evaluator.kwargs['model_kwargs'][slot]['model_id'] = model_id
+                        phy_model.evaluator.kwargs['model_kwargs'][slot]['use_input'] = False
+                        phy_model.evaluator.kwargs['model_kwargs'][slot]['use_output'] = False
                     else:
-                        slot = training_config_slot[model_id]
+                        slot = evaluator_slot[model_id]
                     # If a model's inputs/outputs are not used in the multi-model
                     # the codegen and trainer should not generate and use them
                     # "use_input" and "use_output" are used to mark whether
                     # an input/output of a model is used in a multi-model
                     if new_node.operation.type == '_inputs':
                         input_slot_mapping[new_node] = slot
-                        phy_model.training_config.kwargs['model_kwargs'][slot]['use_input'] = True
+                        phy_model.evaluator.kwargs['model_kwargs'][slot]['use_input'] = True
                     if new_node.operation.type == '_outputs':
                         output_slot_mapping[new_node] = slot
-                        phy_model.training_config.kwargs['model_kwargs'][slot]['use_output'] = True
+                        phy_model.evaluator.kwargs['model_kwargs'][slot]['use_output'] = True
```

Comment:
@hzhua could you please help review this part?

Reply:
No problem if evaluator has the same structure with training_config.